qiita_pet/handlers/api_proxy/prep_template.py

```diff
@@ -6,52 +6,383 @@
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
 from __future__ import division
+import warnings
 
-from qiita_pet.handlers.api_proxy.util import check_access
+from os import remove
+from natsort import natsorted
+
+from qiita_core.util import execute_as_transaction
+from qiita_pet.handlers.api_proxy.util import check_access, check_fp
+from qiita_db.metadata_template.util import load_template_to_dataframe
+from qiita_db.util import convert_to_id
 from qiita_db.study import Study
+from qiita_db.ontology import Ontology
+from qiita_db.metadata_template.prep_template import PrepTemplate
+
+
+@execute_as_transaction
+def _process_investigation_type(inv_type, user_def_type, new_type):
+    """Return the investigation_type and add it to the ontology if needed
+
+    Parameters
+    ----------
+    inv_type : str
+        The investigation type
+    user_def_type : str
+        The user-defined investigation type
+    new_type : str
+        The new user-defined investigation_type
+
+    Returns
+    -------
+    str
+        The investigation type chosen by the user
+    """
+    if inv_type == '':
+        inv_type = None
+    elif inv_type == 'Other' and user_def_type == 'New Type':
+        # This is a new user defined investigation type so store it
+        inv_type = new_type
+        ontology = Ontology(convert_to_id('ENA', 'ontology'))
+        ontology.add_user_defined_term(inv_type)
+    elif inv_type == 'Other' and user_def_type != 'New Type':
```

Comment:
This segment of code doesn't get tested (as far as I know).  Does this need a test?